lib/libzfs/libzfs_dataset.c

```diff
@@ -1249,15 +1256,17 @@ zfs_valid_proplist(libzfs_handle_t *hdl, zfs_type_t type, nvlist_t *nvl,
 			}
 			if (intval != 0 &&
 			    (intval < SPA_MINBLOCKSIZE ||
-			    intval > SPA_OLD_MAXBLOCKSIZE || !ISP2(intval))) {
+			    intval > maxbs || !ISP2(intval))) {
+				zfs_nicebytes(maxbs, buf, sizeof (buf));
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "invalid '%s=%d' property: must be zero or "
-				    "a power of 2 from 512B to 128K"), propname,
```

Comment:
NIT -- I'm wondering if we might consider renaming the property by adding an alias to the zfs cli?  **Small** perhaps is losing it's meaning here  and `special_file_block_limit` might make more sense?

Reply:
`special_recordsize_threshold` ?

Reply:
@kpande I get your way of thinking, but the term recordsize already confuses people... 
it might have some side effects in people's brains to make it: `special_recordsize_threshold`

Reply:
i am not responsible for people's confusion, it *is* a recordsize limit and not a block limit, since zvol are unaffected by this. you can make it an easier to remember name by removing 'files_block' and using 'recordsize'.

Reply:
Let's leave adding an alias to another PR if we decide it's worthwhile.